#!/bin/sh
# This is a shell archive (produced by shar 3.50)
# To extract the files from this archive, save it to a file, remove
# everything above the "!/bin/sh" line above, and type "sh file_name".
#
# made 07/28/1992 13:57 UTC by aw@jello
# Source directory /fs/11/m/acp/aw/Xbae-rls/Xbae-7-public/patch
#
# existing files will NOT be overwritten unless -c is specified
#
# This shar contains:
# length  mode       name
# ------ ---------- ------------------------------------------
#    380 -rw-r--r-- README.patch
#  11929 -rw-r--r-- patch-3.6-to-3.7
#
# ============= README.patch ==============
if test -f 'README.patch' -a X"$1" != X"-c"; then
	echo 'x - skipping README.patch (File already exists)'
else
echo 'x - extracting README.patch (Text)'
sed 's/^X//' << 'SHAR_EOF' > 'README.patch' &&
This patch brings Xbae 3.6 up to version 3.7.  See the patched
README for what was changed.  This patch does *not* patch the
PostScript documents.  If you want the new docs, ftp the
entire 3.7 distribution from export.lcs.mit.edu in
contrib/Xbae-widgets-3.7.tar.Z.
X
To apply the patch, cd to the toplevel directory in the Xbae
distribution, and do
X
X  patch -p0 < patch-3.6-to-3.7
SHAR_EOF
chmod 0644 README.patch ||
echo 'restore of README.patch failed'
Wc_c="`wc -c < 'README.patch'`"
test 380 -eq "$Wc_c" ||
	echo 'README.patch: original size 380, current size' "$Wc_c"
fi
# ============= patch-3.6-to-3.7 ==============
if test -f 'patch-3.6-to-3.7' -a X"$1" != X"-c"; then
	echo 'x - skipping patch-3.6-to-3.7 (File already exists)'
else
echo 'x - extracting patch-3.6-to-3.7 (Text)'
sed 's/^X//' << 'SHAR_EOF' > 'patch-3.6-to-3.7' &&
*** ../../Xbae-6-public/Xbae-3.6/README	Fri Apr 10 10:47:44 1992
--- README	Wed Jul  8 15:53:08 1992
***************
*** 1,5 ****
! Xbae Library Version 3.6 (see src/version.c)
! Xbae 3.6 is the second public release.
X  
X  The Xbae library is a set of two Motif widgets, XbaeMatrix and
X  XbaeCaption.  It was developed using X11R4/X11R5 and Motif 1.1.[0-4]
--- 1,5 ----
! Xbae Library Version 3.7 (see src/version.c)
! Xbae 3.7 is the third public release.
X  
X  The Xbae library is a set of two Motif widgets, XbaeMatrix and
X  XbaeCaption.  It was developed using X11R4/X11R5 and Motif 1.1.[0-4]
***************
*** 51,56 ****
--- 51,68 ----
X     "Group Box", but with more flexibility.  XbaeCaption is also useful
X     for associating labels with individual XmTextField widgets.
X  
+ Bugs fixed in version 3.7:
+ 
+  * XbaeMatrix would occasionally steal the focus from other widgets
+    in the same shell.  The Clip widget was changed to only respond
+    to synthetic FocusIn events.
+  * XbaeMatrix StringToAlignmentArray converter should free the array
+    when the conversion fails.
+  * All header files now use a more specific inclusion protection symbol
+    to avoid conflicts with other packages.
+  * The XbaeMatrix documentation incorrectly documented the
+    XbaeMatrixLeaveCellCallbackStruct.
+  
X  Bugs fixed in version 3.6:
X  
X   * XbaeMatrix computes a bad width when XmNvisibleColumns == XmNcolumns.
*** ../../Xbae-6-public/Xbae-3.6/src/Caption.c	Thu Apr  9 11:25:46 1992
--- src/Caption.c	Wed Jul  8 15:43:12 1992
***************
*** 34,40 ****
X  #include <Xbae/CaptionP.h>
X  
X  #ifndef SABER
! static char sccsid[] = "@(#)Caption.c	1.7 3/17/92";
X  #endif
X  
X  
--- 34,40 ----
X  #include <Xbae/CaptionP.h>
X  
X  #ifndef SABER
! static char sccsid[] = "@(#)Caption.c	1.8 7/8/92";
X  #endif
X  
X  
***************
*** 166,172 ****
X  	/* extension			*/  NULL
X      },
X      {
!     /* matrix_class fields */
X  	/* extension			*/ NULL,
X      }
X  };
--- 166,172 ----
X  	/* extension			*/  NULL
X      },
X      {
!     /* caption_class fields */
X  	/* extension			*/ NULL,
X      }
X  };
***************
*** 234,240 ****
X       */
X      (void)
X  	XtVaCreateManagedWidget(XtName(new),
! 				xmLabelWidgetClass, new,
X  				XmNbackground,	new->core.background_pixel,
X  				XmNforeground,	new->manager.foreground,
X  				XmNfontList,	new->caption.font_list,
--- 234,240 ----
X       */
X      (void)
X  	XtVaCreateManagedWidget(XtName(new),
! 				xmLabelWidgetClass, (Widget)new,
X  				XmNbackground,	new->core.background_pixel,
X  				XmNforeground,	new->manager.foreground,
X  				XmNfontList,	new->caption.font_list,
*** ../../Xbae-6-public/Xbae-3.6/src/Caption.h	Thu Apr  9 11:25:46 1992
--- src/Caption.h	Wed Jul  8 15:43:13 1992
***************
*** 22,32 ****
X   * CaptionWidget Author: Andrew Wason, Bellcore, aw@bae.bellcore.com
X   */
X  
! #ifndef _Caption_h
! #define _Caption_h
X  
X  /*
!  *  "@(#)Caption.h	1.4 2/18/92"
X   */
X  
X  /*
--- 22,32 ----
X   * CaptionWidget Author: Andrew Wason, Bellcore, aw@bae.bellcore.com
X   */
X  
! #ifndef _Xbae_Caption_h
! #define _Xbae_Caption_h
X  
X  /*
!  *  "@(#)Caption.h	1.5 7/8/92"
X   */
X  
X  /*
***************
*** 91,96 ****
X      XbaeAlignmentBottomOrRight
X  } XbaeLabelAlignment;
X  
! #endif /* _Caption_h */
X  /* DON'T ADD STUFF AFTER THIS #endif */
X  
--- 91,96 ----
X      XbaeAlignmentBottomOrRight
X  } XbaeLabelAlignment;
X  
! #endif /* _Xbae_Caption_h */
X  /* DON'T ADD STUFF AFTER THIS #endif */
X  
*** ../../Xbae-6-public/Xbae-3.6/src/CaptionP.h	Thu Apr  9 11:25:47 1992
--- src/CaptionP.h	Wed Jul  8 15:43:13 1992
***************
*** 26,36 ****
X   * CaptionP.h - Private definitions for Caption widget
X   */
X  
! #ifndef _CaptionP_h
! #define _CaptionP_h
X  
X  /*
!  *  "@(#)CaptionP.h	1.4 2/18/92"
X   */
X  
X  #include <Xm/XmP.h>
--- 26,36 ----
X   * CaptionP.h - Private definitions for Caption widget
X   */
X  
! #ifndef _Xbae_CaptionP_h
! #define _Xbae_CaptionP_h
X  
X  /*
!  *  "@(#)CaptionP.h	1.5 7/8/92"
X   */
X  
X  #include <Xm/XmP.h>
***************
*** 86,89 ****
X      XbaeCaptionPart	caption;
X  } XbaeCaptionRec;
X  
! #endif /* _CaptionP_h */
--- 86,89 ----
X      XbaeCaptionPart	caption;
X  } XbaeCaptionRec;
X  
! #endif /* _Xbae_CaptionP_h */
*** ../../Xbae-6-public/Xbae-3.6/src/Clip.c	Thu Apr  9 11:26:16 1992
--- src/Clip.c	Wed Jul  8 15:42:52 1992
***************
*** 31,37 ****
X  #include <Xbae/ClipP.h>
X  
X  #ifndef SABER
! static char sccsid[] = "@(#)Clip.c	3.3 2/18/92";
X  #endif
X  
X  static char defaultTranslations[] =
--- 31,37 ----
X  #include <Xbae/ClipP.h>
X  
X  #ifndef SABER
! static char sccsid[] = "@(#)Clip.c	3.4 5/13/92";
X  #endif
X  
X  static char defaultTranslations[] =
***************
*** 214,220 ****
X  String *params;
X  Cardinal *nparams;
X  {
!     if (event->xany.type != FocusIn)
X  	return;
X  
X      if (cw->clip.focus_callback)
--- 214,220 ----
X  String *params;
X  Cardinal *nparams;
X  {
!     if (event->xany.type != FocusIn || !event->xfocus.send_event)
X  	return;
X  
X      if (cw->clip.focus_callback)
*** ../../Xbae-6-public/Xbae-3.6/src/Clip.h	Thu Apr  9 11:26:16 1992
--- src/Clip.h	Wed Jul  8 15:42:53 1992
***************
*** 26,36 ****
X   * Clip.h - Public definitions for Clip widget
X   */
X  
! #ifndef _Clip_h
! #define _Clip_h
X  
X  /*
!  *  "@(#)Clip.h	3.3 2/18/92"
X   */
X  
X  #include <Xm/Xm.h>
--- 26,36 ----
X   * Clip.h - Public definitions for Clip widget
X   */
X  
! #ifndef _Xbae_Clip_h
! #define _Xbae_Clip_h
X  
X  /*
!  *  "@(#)Clip.h	3.4 7/8/92"
X   */
X  
X  #include <Xm/Xm.h>
***************
*** 73,76 ****
X  }
X  #endif
X  
! #endif /* _Clip_h */
--- 73,76 ----
X  }
X  #endif
X  
! #endif /* _Xbae_Clip_h */
*** ../../Xbae-6-public/Xbae-3.6/src/ClipP.h	Thu Apr  9 11:26:16 1992
--- src/ClipP.h	Wed Jul  8 15:42:53 1992
***************
*** 26,36 ****
X   * ClipP.h - Private definitions for Clip widget
X   */
X  
! #ifndef _ClipP_h
! #define _ClipP_h
X  
X  /*
!  *  "@(#)ClipP.h	3.3 2/18/92"
X   */
X  
X  #include <Xm/XmP.h>
--- 26,36 ----
X   * ClipP.h - Private definitions for Clip widget
X   */
X  
! #ifndef _Xbae_ClipP_h
! #define _Xbae_ClipP_h
X  
X  /*
!  *  "@(#)ClipP.h	3.4 7/8/92"
X   */
X  
X  #include <Xm/XmP.h>
***************
*** 96,99 ****
X      XbaeClipPart	clip;
X  } XbaeClipRec;
X  
! #endif /* _ClipP_h */
--- 96,99 ----
X      XbaeClipPart	clip;
X  } XbaeClipRec;
X  
! #endif /* _Xbae_ClipP_h */
*** ../../Xbae-6-public/Xbae-3.6/src/Matrix.h	Thu Apr  9 11:26:22 1992
--- src/Matrix.h	Wed Jul  8 15:42:51 1992
***************
*** 22,32 ****
X   * MatrixWidget Author: Andrew Wason, Bellcore, aw@bae.bellcore.com
X   */
X  
! #ifndef _Matrix_h
! #define _Matrix_h
X  
X  /*
!  *  "@(#)Matrix.h	3.7 3/11/92"
X   */
X  
X  /*
--- 22,32 ----
X   * MatrixWidget Author: Andrew Wason, Bellcore, aw@bae.bellcore.com
X   */
X  
! #ifndef _Xbae_Matrix_h
! #define _Xbae_Matrix_h
X  
X  /*
!  *  "@(#)Matrix.h	3.8 7/8/92"
X   */
X  
X  /*
***************
*** 371,376 ****
X  } XbaeMatrixSelectCellCallbackStruct;
X  
X  
! #endif /* _Matrix_h */
X  /* DON'T ADD STUFF AFTER THIS #endif */
X  
--- 371,376 ----
X  } XbaeMatrixSelectCellCallbackStruct;
X  
X  
! #endif /* _Xbae_Matrix_h */
X  /* DON'T ADD STUFF AFTER THIS #endif */
X  
*** ../../Xbae-6-public/Xbae-3.6/src/MatrixP.h	Thu Apr  9 11:26:22 1992
--- src/MatrixP.h	Wed Jul  8 15:42:52 1992
***************
*** 26,36 ****
X   * MatrixP.h - Private definitions for Matrix widget
X   */
X  
! #ifndef _MatrixP_h
! #define _MatrixP_h
X  
X  /*
!  *  "@(#)MatrixP.h	3.8 4/8/92"
X   */
X  
X  #include <Xm/XmP.h>
--- 26,36 ----
X   * MatrixP.h - Private definitions for Matrix widget
X   */
X  
! #ifndef _Xbae_MatrixP_h
! #define _Xbae_MatrixP_h
X  
X  /*
!  *  "@(#)MatrixP.h	3.11 7/8/92"
X   */
X  
X  #include <Xm/XmP.h>
***************
*** 415,418 ****
X      XbaeMatrixPart	matrix;
X  } XbaeMatrixRec;
X  
! #endif /* _MatrixP_h */
--- 415,418 ----
X      XbaeMatrixPart	matrix;
X  } XbaeMatrixRec;
X  
! #endif /* _Xbae_MatrixP_h */
*** ../../Xbae-6-public/Xbae-3.6/src/version.c	Thu Apr  9 11:26:22 1992
--- src/version.c	Wed Jul  8 16:28:45 1992
***************
*** 21,27 ****
X   */
X  
X  #ifndef SABER
! static char sccsid[] = "@(#)version.c	3.6 4/9/92";
X  #endif
X  
X  /*
--- 21,27 ----
X   */
X  
X  #ifndef SABER
! static char sccsid[] = "@(#)version.c	3.7 7/8/92";
X  #endif
X  
X  /*
***************
*** 29,42 ****
X    This file contains version information for a given Xbae release.
X    The SCCSID of this file should be used as the Xbae release number.
X  
!   Matrix.h	3.7
!   MatrixP.h	3.8
!   Matrix.c	3.16
!   Clip.h	3.3
!   ClipP.h	3.3
!   Clip.c	3.3
!   Caption.h	1.4
!   CaptionP.h	1.4
!   Caption.c	1.7
X  
X  */
--- 29,46 ----
X    This file contains version information for a given Xbae release.
X    The SCCSID of this file should be used as the Xbae release number.
X  
!   Matrix.h	3.8
!   MatrixP.h	3.11
!   Matrix.c	3.20
!   Clip.h	3.4
!   ClipP.h	3.4
!   Clip.c	3.4
!   Caption.h	1.5
!   CaptionP.h	1.5
!   Caption.c	1.8
!   // BusyWindow not included in public release
!   BusyWindow.c	1.4
!   BusyWindoP.h	1.3
!   BusyWindow.h	1.3
X  
X  */
*** ../../Xbae-6-public/Xbae-3.6/src/Matrix.c	Thu Apr  9 11:26:20 1992
--- src/Matrix.c	Wed Jul  8 16:28:47 1992
***************
*** 38,44 ****
X  #include <Xbae/MatrixP.h>
X  
X  #ifndef SABER
! static char sccsid[] = "@(#)Matrix.c	3.16 4/9/92";
X  #endif
X  
X  #ifndef XlibSpecificationRelease
--- 38,44 ----
X  #include <Xbae/MatrixP.h>
X  
X  #ifndef SABER
! static char sccsid[] = "@(#)Matrix.c	3.20 7/8/92";
X  #endif
X  
X  #ifndef XlibSpecificationRelease
***************
*** 1044,1050 ****
X       */
X  
X      (void) XtVaCreateWidget("horizScroll",
! 			    xmScrollBarWidgetClass, new,
X  			    XmNorientation,	XmHORIZONTAL,
X  			    XmNdragCallback,	HSCallback,
X  			    XmNvalueChangedCallback, HSCallback,
--- 1044,1050 ----
X       */
X  
X      (void) XtVaCreateWidget("horizScroll",
! 			    xmScrollBarWidgetClass, (Widget)new,
X  			    XmNorientation,	XmHORIZONTAL,
X  			    XmNdragCallback,	HSCallback,
X  			    XmNvalueChangedCallback, HSCallback,
***************
*** 1065,1071 ****
X      HORIZ_ORIGIN(new) = 0;
X  
X      (void) XtVaCreateWidget("vertScroll",
! 			    xmScrollBarWidgetClass, new,
X  			    XmNorientation,		XmVERTICAL,
X  			    XmNdragCallback,		VSCallback,
X  			    XmNvalueChangedCallback,	VSCallback,
--- 1065,1071 ----
X      HORIZ_ORIGIN(new) = 0;
X  
X      (void) XtVaCreateWidget("vertScroll",
! 			    xmScrollBarWidgetClass, 	(Widget)new,
X  			    XmNorientation,		XmVERTICAL,
X  			    XmNdragCallback,		VSCallback,
X  			    XmNvalueChangedCallback,	VSCallback,
***************
*** 1091,1097 ****
X       * Create the Clip widget managed so we can use it for traversal
X       */
X      (void) XtVaCreateManagedWidget("clip",
! 				   xbaeClipWidgetClass, new,
X  				   XmNexposeProc,	ClipRedisplay,
X  				   XmNtraversalOn,
X  				   	new->manager.traversal_on,
--- 1091,1097 ----
X       * Create the Clip widget managed so we can use it for traversal
X       */
X      (void) XtVaCreateManagedWidget("clip",
! 				   xbaeClipWidgetClass, (Widget)new,
X  				   XmNexposeProc,	ClipRedisplay,
X  				   XmNtraversalOn,
X  				   	new->manager.traversal_on,
***************
*** 1111,1117 ****
X       * in Realize to be a subwindow of Clip
X       */
X      (void) XtVaCreateWidget("textField",
! 			    xmTextFieldWidgetClass,  new,
X  			    XmNmarginWidth,	new->matrix.cell_margin_width,
X  			    XmNmarginHeight,	new->matrix.cell_margin_height,
X  			    XmNtranslations,	new->matrix.text_translations,
--- 1111,1117 ----
X       * in Realize to be a subwindow of Clip
X       */
X      (void) XtVaCreateWidget("textField",
! 			    xmTextFieldWidgetClass,  (Widget)new,
X  			    XmNmarginWidth,	new->matrix.cell_margin_width,
X  			    XmNmarginHeight,	new->matrix.cell_margin_height,
X  			    XmNtranslations,	new->matrix.text_translations,
***************
*** 7597,7602 ****
--- 7597,7603 ----
X  	    else {
X  		XtDisplayStringConversionWarning(dpy, from->addr,
X  						 XmRAlignmentArray);
+ 		XtFree(array);
X  		return False;
X  	    }
X  
SHAR_EOF
chmod 0644 patch-3.6-to-3.7 ||
echo 'restore of patch-3.6-to-3.7 failed'
Wc_c="`wc -c < 'patch-3.6-to-3.7'`"
test 11929 -eq "$Wc_c" ||
	echo 'patch-3.6-to-3.7: original size 11929, current size' "$Wc_c"
fi
exit 0
